Drivers/Src/Switches.c

```diff
@@ -48,46 +70,117 @@ void Switches_Init(void){
  * @return  ON/OFF State
  */ 
 State Switches_Read(switches_t sw){
-    uint8_t query[3]={SPI_OPCODE_R,SPI_GPIOA,0x00}; //query GPIOA
-    uint8_t SwitchReadData[2] = {0};
+    uint8_t query[3]={SPI_OPCODE_R,SPI_GPIOA, 0}; //query GPIOA
+    uint8_t SwitchReadData[1] = {0};
     // If we are not trying to get the state of the ignition switches, or the Reverse switch
     switch(sw) {
-    case CRUZ_SW:
+    case CRUZ_ST:
     case CRUZ_EN:
-    case HZD_SQ: 
-    case FR_SW: 
+    case REV_SW: 
+    case FOR_SW: 
     case HEADLIGHT_SW: 
     case LEFT_SW: 
     case RIGHT_SW: 
     case REGEN_SW:
+        OSMutexPend(
+            &SwitchMutex,
+            0,
+            OS_OPT_PEND_BLOCKING,
+            &timestamp,
+            &err
+        );
+        if(err != OS_ERR_NONE){
+            return ERROR; //Os error, could not properly lock the Minion SPI Line
+        }
+        GPIO_WriteBit(GPIOA, GPIO_Pin_4, Bit_RESET);
```

Comment:
This function sets the GPIO pin atomically, so you only need the mutexas around this if you want setting all of the GPIO pins together to be atomic

Reply:
The mutex lock is primarily for the SPI send, and I wanted to group the entire chip select - query - chip deselect all in one runthrough